.travis.yml

```diff
@@ -0,0 +1,50 @@
+language: android
+jdk: oraclejdk8
+cache:
+  directories:
+  - "~/arduino_ide"
+  - "~/.arduino15/packages/"
+  - $HOME/.m2
+env:
+  global:
+  - ADB_INSTALL_TIMEOUT=10
+  - ANDROID_TARGET=android-22
+  - CXX=g++-6
+android:
+  components:
+  - tools
+  - tools
+  - platform-tools
+  - build-tools-28.0.3
+  - "$ANDROID_TARGET"
+  - sys-img-armeabi-v7a-$ANDROID_TARGET
+  - extra-android-m2repository
+  - extra-google-m2repository
+compiler:
+- g++
+addons:
+  apt:
+    sources:
+    - ubuntu-toolchain-r-test
+    - llvm-toolchain-precise-3.8
+    packages:
+    - g++-6
+before_script:
+- android list targets
+- echo no | android create avd --force -n test -t $ANDROID_TARGET --abi armeabi-v7a
+- emulator -avd test -no-audio -no-window &
+- android-wait-for-emulator
+- adb devices
+- adb shell input keyevent 82 &
+- chmod +x $TRAVIS_BUILD_DIR/Android_Application_for_SmartCar/gradlew
+before_install:
+- echo yes | sdkmanager "platforms;android-28"
+- source <(curl -SLs https://raw.githubusercontent.com/adafruit/travis-ci-arduino/master/install.sh)
+install:
+- arduino --install-library "Smartcar shield"
+script:
+- cd $TRAVIS_BUILD_DIR/Android_Application_for_SmartCar
+- "./gradlew assembleDebug"
+- ls $TRAVIS_BUILD_DIR/Android_Application_for_SmartCar/app/build/apk/
```

Comment:
Your `.apk` is most likely located at `$TRAVIS_BUILD_DIR/Android_Application_for_SmartCar/app/build/outputs/apk/debug/app-debug.apk`.

Reply:
Thank you for the information, I am still having trouble with keeping the file after the script ends. Apparently there's a cleanup process that deletes it.